#!/usr/bin/env python
PACKAGE = "imagine_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

M_PI = 3.14159265359

gen.add("width", int_t, 4, "Width", 200, 1, 1000)
gen.add("height", int_t, 4, "Height", 150, 1, 1000)

gen.add("range_min", double_t, 4, "Minimum range", 0.0, 0.0, 10.0)
gen.add("range_max", double_t, 4, "Maximum range", 100.0, 0.0, 1000.0)

gen.add("cx", double_t, 4, "Cx", 100.0, 0.0, 1000.0)
gen.add("cy", double_t, 4, "Cy", 75.0, 0.0, 1000.0)

gen.add("fx", double_t, 4, "Fx", 100.0, 0.0, 2000.0)
gen.add("fy", double_t, 4, "Fy", 100.0, 0.0, 2000.0)

exit(gen.generate(PACKAGE, "camera_simulator", "CameraModel"))